src/metpy/xarray.py

```diff
@@ -1172,10 +1172,18 @@ def preprocess_and_wrap(broadcast=None, wrap_like=None, match_unit=False, to_mag
     def decorator(func):
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
-            bound_args = signature(func).bind(*args, **kwargs)
+            sig = signature(func)
+            bound_args = sig.bind(*args, **kwargs)
 
             # Auto-broadcast select xarray arguments, and update bound_args
             if broadcast is not None:
+                for arg_name in broadcast:
+                    if arg_name not in sig.parameters:
+                        raise ValueError(
+                            f'Cannot broadcast argument {arg_name} as it is not in function '
+                            'signature'
+                        )
+
```

Comment:
That means this is happening at function execution time, right? Couldn't this instead be at the top of `decorator` since it never depends on what's being passed to the wrapper?

Reply:
Ah, good point! I missed that since I originally checked against `bound_args.arguments` (but that failed due to optional arguments). I'll make that change.